src/modules/partition/gui/PartitionLabelsView.cpp

```diff
@@ -190,6 +192,11 @@ PartitionLabelsView::buildTexts( const QModelIndex& index ) const
             {
                 firstLine = tr( "EFI system", "@label" );
             }
+            else if ( index.data( PartitionModel::FileSystemTypeRole ).toInt() == FileSystem::Unformatted
+                      && static_cast< Partition* >( index.data( PartitionModel::PartitionPtrRole ).value< void* >() )->property( "_calamares_flags" ).toUInt() & KPM_PARTITION_FLAG( BiosGrub ) )
```

Comment:
Can we give all that casting-and-wrangling a nicer name, make it a static function in this translation unit?

Reply:
Will do.